surf/apps/materials/models.py

```diff
@@ -95,6 +95,27 @@ class Material(UUIDModel):
     keywords = django_models.TextField(blank=True, null=True)
     deleted_at = django_models.DateTimeField(null=True)
 
+    star_1 = django_models.PositiveIntegerField(default=0)
+    star_2 = django_models.PositiveIntegerField(default=0)
+    star_3 = django_models.PositiveIntegerField(default=0)
+    star_4 = django_models.PositiveIntegerField(default=0)
+    star_5 = django_models.PositiveIntegerField(default=0)
+    # a user can applaud a material many times more than he can star rate it
+    applaud_count = django_models.BigIntegerField(default=0)
+    view_count = django_models.BigIntegerField(default=0)
+
+    def get_star_count(self):
+        return self.star_1 + self.star_2 + self.star_3 + self.star_4 + self.star_5
+    get_star_count.short_description = 'Star rating count'
+
+    def get_avg_star_rating(self):
+        total_stars = 1 * self.star_1 + 2 * self.star_2 + 3 * self.star_3 + 4 * self.star_4 + 5 * self.star_5
+        star_count = self.get_star_count()
+        if star_count > 0:
+            return round(total_stars / star_count, 1)
```

Comment:
I decided to round here instead of in the front-end because javascript round is terrible, however not 100% sure about using 1 decimal instead of 0, I just thought it looked nicer in the frontend.

Reply:
Ok, check